internal/support/notifications/v2/application/subscription.go

```diff
@@ -128,3 +130,45 @@ func DeleteSubscriptionByName(name string, ctx context.Context, dic *di.Containe
 	}
 	return nil
 }
+
+// PatchSubscription executes the PATCH operation with the subscription DTO to replace the old data
+func PatchSubscription(ctx context.Context, dto dtos.UpdateSubscription, dic *di.Container) errors.EdgeX {
+	dbClient := v2NotificationsContainer.DBClientFrom(dic.Get)
+	lc := container.LoggingClientFrom(dic.Get)
+
+	var subscription models.Subscription
+	var edgexErr errors.EdgeX
+	if dto.Id != nil {
+		subscription, edgexErr = dbClient.SubscriptionById(*dto.Id)
+		if edgexErr != nil {
+			return errors.NewCommonEdgeXWrapper(edgexErr)
+		}
+	} else {
+		subscription, edgexErr = dbClient.SubscriptionByName(*dto.Name)
+		if edgexErr != nil {
+			return errors.NewCommonEdgeXWrapper(edgexErr)
+		}
+	}
+	if dto.Name != nil && *dto.Name != subscription.Name {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("subscription name '%s' not match the existing '%s' ", *dto.Name, subscription.Name), nil)
+	}
+
+	requests.ReplaceSubscriptionModelFieldsWithDTO(&subscription, dto)
+	_, edgeXerr := dbClient.SubscriptionByName(subscription.Name)
+	if edgeXerr != nil {
+		return errors.NewCommonEdgeX(errors.Kind(edgeXerr), fmt.Sprintf("subscription '%s' existence check failed", subscription.Name), edgeXerr)
+	}
```

Comment:
Why is this check needed? Doesn't the code above fail if the `name`  doesn't exist or match? and the Delete below will fail if those above don't.

Reply:
Yep, you are right. Removed.